lib/streamlit/runtime/caching/cache_utils.py

```diff
@@ -602,6 +616,9 @@ def save_element_message(
             else:
                 widget_meta = None
 
+            image_data = self._image_data
+            self._image_data = []
```

Comment:
Why are we setting the image_data and then resetting it?

Reply:
We're saving the reference locally so it can be saved in the cache, then resetting the instance field. I could move the resetting part down so it's clearer that it is side-cause cleanup.